runtime/vm/jvminit.c

```diff
@@ -2606,34 +2606,34 @@ modifyDllLoadTable(J9JavaVM * vm, J9Pool* loadTable, J9VMInitArgs* j9vm_args)
 
 		if (strcmp(testString, VMOPT_XINT)==0) {
 			if (xjit || xnojit || xnoaot || xaot) {
-				SET_FLAG_AT_INDEX( NOT_CONSUMABLE_ARG, i );
+				SET_FLAG_AT_INDEX( NOT_CONSUMABLE_ARG | (j9vm_args->j9Options[i].flags & USER_ARG), i );
 			} else {
 				xint = TRUE;
 			}
 		} else if ((strcmp(testString, VMOPT_XJIT)==0) ||
 					(strstr(testString, VMOPT_XJIT_COLON)==testString)) {
 			if (xint || xnojit) {
```

Comment:
Can you grab `(j9vm_args->j9Options[i].flags & USER_ARG)` into a local variable and use that instead?